waffel/waffel.py

```diff
@@ -21,112 +22,165 @@
 #  along with eis-waffel. If not, see <http://www.gnu.org/licenses/>.
 #
 
-#***************************************
-# App
+# ***************************************
+
+from datetime import datetime
+from datetime import timedelta
+
+import dateutil.parser
+import logging
 import requests
 
 
-class Waffel:
-    # due to historic reasons music is named arts and arts is named campus in EIS
-    TRACK_NAME_MAP = {"art": "music", "campus": "arts"}
+logging.basicConfig()  # TODO: changed print to logging.
+logger = logging.getLogger(__file__)
 
-    def __init__(self, api_url, timeout=30):
+
+class Waffel(object):
+
+    def __init__(
+            self,
+            api_url,
+            tracks,
+            locations,
+            timezone='Europe/Vienna',
+            timeout=30):
         self.api_url = api_url
+        self.track_map = {it['eis_id']: it['id'] for it in tracks}
+        self.location_map = {it['eis_id']: it['id'] for it in locations}
+        self.timezone = timezone
         self.timeout = timeout
-        self.headers = []
+        self.year = datetime.now().year
+        self.headers = {'Accept': 'application/json; charset=utf-8'}
+        self.min_delta = timedelta(minutes=30)
+        self.max_delta = timedelta(hours=12)
+
+    # due to historic reasons music is named arts and arts is named campus in EIS  # noqa
+    TRACK_NAME_MAP = {"art": "music", "campus": "arts"}
 
```

Comment:
TRACK_NAME_MAP is not needed anymore since we do this via config['tracks']

Reply:
done